aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,398 @@
+      // if it is not a list all op, just check if the namespace exists and return itself.
+      if (namespaceExists(namespace)) {
+        return Lists.newArrayList(namespace);
+      }
+      throw new NoSuchNamespaceException(
+          "Glue does not support nested namespace, cannot list namespaces under %s", namespace);
+    }
+    // should be safe to list all before returning the list, instead of dynamically load the list.
+    String nextToken = null;
+    List<Namespace> results = new ArrayList<>();
+    do {
+      GetDatabasesResponse response = glue.getDatabases(GetDatabasesRequest.builder()
+          .catalogId(catalogId)
+          .nextToken(nextToken)
+          .build());
+      if (response.hasDatabaseList()) {
+        results.addAll(response.databaseList().stream()
+            .map(GlueToIcebergConverter::toNamespace)
+            .collect(Collectors.toList()));
+        nextToken = response.nextToken();
+      }
+    } while (nextToken != null);
+    LOG.debug("Listing namespace {} returned namespaces: {}", namespace, results);
+    return results;
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) throws NoSuchNamespaceException {
+    String databaseName = IcebergToGlueConverter.toDatabaseName(namespace);
+    try {
+      GetDatabaseResponse response = glue.getDatabase(GetDatabaseRequest.builder()
+          .catalogId(catalogId)
+          .name(databaseName)
+          .build());
+      Map<String, String> result = response.database().parameters();
+      LOG.debug("Loaded metadata for namespace {} found {}", namespace, result);
+      return result;
+    } catch (InvalidInputException e) {
+      throw new NoSuchNamespaceException("invalid input for namespace %s, error message: %s",
+          namespace, e.getMessage());
+    } catch (EntityNotFoundException e) {
+      throw new NoSuchNamespaceException("fail to find Glue database for namespace %s, error message: %s",
+          databaseName, e.getMessage());
+    }
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) throws NamespaceNotEmptyException {
+    namespaceExists(namespace);
+    List<TableIdentifier> tableIdentifiers = listTables(namespace);
+    if (!CollectionUtils.isEmpty(tableIdentifiers)) {
+      throw new NamespaceNotEmptyException("Cannot drop namespace %s because it is not empty. " +
+          "The following tables still exist under the namespace: %s", namespace, tableIdentifiers);
+    }
+    glue.deleteDatabase(DeleteDatabaseRequest.builder()
+        .catalogId(catalogId)
+        .name(IcebergToGlueConverter.toDatabaseName(namespace))
+        .build());
+    LOG.info("Dropped namespace: {}", namespace);
+    // Always successful, otherwise exception is thrown
+    return true;
+  }
+
+  @Override
+  public boolean setProperties(Namespace namespace, Map<String, String> properties) throws NoSuchNamespaceException {
+    Map<String, String> parameter = Maps.newHashMap();
```

Comment:
This name isn't very descriptive. It seems to mean "a parameter that will be passed somewhere", but doesn't describe what the object is. I think a better name would be `newProperties` or similar. That fits the conventions we use elsewhere.

Reply:
The method below uses `metadata`, which is also good.

Reply:
Sure, `parameter` is used to mean `properties` in Glue that's why I used it. Will update to that.